Tools/ardupilotwaf/cxx_checks.py

```diff
@@ -157,26 +157,46 @@ def check_librt(cfg, env):
     return ret
 
 @conf
+def check_package(cfg, env, libname):
+    '''use pkg-config to look for an installed library that has a LIBNAME.pc file'''
+    cfg.check_cfg(package=libname, mandatory=False, global_define=True,
+                  args=['--libs', '--cflags'])
+    capsname = libname.upper()
+    env.LIB += cfg.env['LIB_%s' % capsname]
+    env.INCLUDES += cfg.env['INCLUDES_%s' % capsname]
+    env.CFLAGS += cfg.env['CFLAGS_%s' % capsname]
+    env.LIBPATH += cfg.env['LIBPATH_%s' % capsname]
```

Comment:
https://gist.github.com/lucasdemarchi/9fed061f028e9dfeed053be980f5bece

This seems to work. The difference on these approaches are: you are telling everyone to link and use these libraries. With the uselib approach we can control which binary/lib need to link to them.  We'd still probably need to pass these `uselib=` to ap_stlib, and examples.